### ISSUE TITLE
Multiple UPE - Resolve failing tests and checks

### ISSUE THREAD

**Issue by FangedParakeet** [Automattic/woocommerce-payments#4907]
### Description
The unit tests on the current working branch have been failing for quite a long time: I didn't update or create any new unit tests when I first created the POC branch and I'm not too sure that any have been updated since. Nonetheless the split-UPE fundamentally changes the checkout layout and the ways in which the payment gateway classes are utilised.

We need to update these tests and ensure that are no failures before we can eventually merge this branch back into `develop`. If we could be more confident in our testing environment, I think we could all sleep a little sounder in the knowledge that most user-flows and edge-cases have some form of acceptable test coverage.

### Acceptance criteria

* Ensure no more test failures on GH.

N.B. The `_wcpay_feature_upe` feature flag will soon be retired, since in the split-UPE approach there will not need to be a distinction between legacy and "new" checkout experiences (see #4724). Currently, existing tests are generally split into separate test-cases for when the UPE is enabled and disabled. We will need to resolve and consolidate these diverging tests into compatible tests for our combined single checkout experience.

**Comment by FangedParakeet** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/Automattic/woocommerce-payments/4907?planning-poker) with Zenhub @mdmoore @timur27

**Comment by timur27** [comment]
Thanks for creating the issue, @FangedParakeet! 🚀 
Are we talking about e2e tests failing ([1](https://github.com/Automattic/woocommerce-payments/actions/runs/3242683626/jobs/5316304675), [2](https://github.com/Automattic/woocommerce-payments/actions/runs/3242683626/jobs/5316305260), [3](https://github.com/Automattic/woocommerce-payments/actions/runs/3242683626/jobs/5316305120), [4](https://github.com/Automattic/woocommerce-payments/actions/runs/3242683626/jobs/5316304990) and [5](https://github.com/Automattic/woocommerce-payments/actions/runs/3242683626/jobs/5316304816)), or there are unit tests which need to be aligned? I didn't find anything besides the E2E tests failing on [the branch](https://github.com/Automattic/woocommerce-payments/pull/3968), which is why I wanted to double-check. 

Thank you!

**Comment by FangedParakeet** [comment]
I'm actually quite pleasantly surprised that we haven't broken any unit tests! Pontificating more on this perplexion, perhaps that isn't so perplexing after all, since we aren't really changing the payments flow and we do have passable payment method test coverage already that remains mostly unaffected.

In that case, the baleful gaze of this issue revolves towards the shadowy realm of the E2E tests instead. @timur27, you  mentioned earlier that you'd spied tests failing on `develop` as well: is that still the case? Let me know if you think that might temporarily block our efforts on this issue. Cheers!

**Comment by mdmoore** [comment]
I think we have @harriswong to thank for [fixing up the unit tests](https://github.com/Automattic/woocommerce-payments/pull/3968/commits/c02dd9af32070ec2a25825fd002f157fad791d6a) a couple of months ago. Glad to see they're still in good shape.

**Comment by timur27** [comment]
> @timur27, you mentioned earlier that you'd spied tests failing on develop as well: is that still the case? 

@FangedParakeet https://github.com/Automattic/woocommerce-payments/pull/5043 issue, which covers failing compatibility tests was fixed by https://github.com/Automattic/woocommerce-payments/pull/5049, so we shouldn't face any failing tests (except those caused by the split UPE) and thus there shouldn't be any blockers down the road. Thanks for keeping the finger on the pulse here!

**Comment by timur27** [comment]
I'd like to give a quick update on this issue. My initial plan was to finish the issue today (Fri 25 Nov), but as I spent some time configuring my local e2e environment for the first time and helped GlobalStep with the testing, I did not finish it yet. I'm targeting to open a PR on Monday. 

This issue consists of three parts: 
1. Fix e2e tests for blocks checkout failing (fixed [here](https://app.zenhub.com/workspaces/heisenberg-63586f2ec60359001310fa6a/issues/automattic/woocommerce-payments/5168))
2. Fix [e2e tests for subscriptions](https://github.com/Automattic/woocommerce-payments/actions/runs/3539079785/jobs/5940585071) (in progress) 
3. Extend the existing tests suite with new ones to cover split UPE critical scenarios with tests (in progress).
